### PR TITLE
[pytorch_gluon cheatsheet] Add sections `Pytorch optimizer vs Gluon Trainer` and `Monitoring`

### PR DESCRIPTION
**Pull request by ZiyueHuang**
Add two sections of `Pytorch optimizer vs Gluon Trainer` and `Monitoring` in pytorch_gluon cheatsheet.

Please notice that the subsection of `zero_grad for grad_req is 'add'` is removed, because I think this is not commonly used in training. Feel free to ask me to add this subsection back if you think it is neccessary.

cc @szha

### CODE REVIEW

File: cheatsheets/pytorch_gluon.md [hunk 1]
```diff
@@ -148,14 +148,41 @@ SymbolBlock can construct block from symbol. This is useful for using pre-traine
...
+| distributed data parallelism |   `torch.distributed.init_process_group(...)`<br/>`model = torch.nn.parallel.distributedDataParallel(model, ...)`    | `store = kv.create('dist')`<br/>`trainer = gluon.Trainer(net.collect_params(), ..., kvstore=store)`  |
+
 ## Monitoring
 ### MXNet has pre-defined metrics
+
+Gluon provide several predefined metrics which can online evaluate the performance of a learned model.
+
+| Function               | Pytorch                           | MXNet Gluon                              |
+|------------------------|-----------------------------------|------------------------------------------|
+| metric |  Not available   | `with autograd.record():`<br/>&nbsp;&nbsp;&nbsp;&nbsp;`output = net(data)`<br/>&nbsp;&nbsp;&nbsp;&nbsp;`L = loss(ouput, label)`<br/>&nbsp;&nbsp;&nbsp;&nbsp;`loss(ouput, label).backward()`<br/>`trainer.step(batch_size)`<br/>`metric.update(label, output)`  |
```
Comment (szha) [inline cheatsheets/pytorch_gluon.md]: Consider adding how `metric` is created.

**Comment by szha** [comment]
Thanks a lot for adding these, @ZiyueHuang. The rest looks good to me.

**Comment by zackchase** [comment]
Thanks @szha and @ZiyueHuang !
